ci/ci-common.yml

```diff
@@ -29,7 +29,7 @@ stages:
     reports:
       dotenv: build.env
   variables:
-    SPACK_SHA: develop-2024-06-02
+    SPACK_SHA: 05c7ff4595a4e574cbcf45475c626b9b94c22af1
     SPACK_BUILDCACHE: develop-2024-06-02
```

Comment:
Do you need/want to update the cache as well to last sunday's cache? Or wait until next sunday?

Reply:
I'm probably missing something, but this is the last version showing up on https://cache.spack.io. Do you know if despite this, a cache is also available for the latest `develop-*` tag?

Reply:
According to people closer to the matter (thanks @msimberg for asking!), it seems that `develop-2024-06-09` did run, but the later pipelines did not. Should we stick with this one for the time being and update it later?

Reply:
Just checking: isn't this working or isn't this enough?  https://cache.spack.io/tag/develop-2024-06-16/

Reply:
It wasn't there yesterday... :eyes:

Reply:
I can now see `develop-2024-06-09` too on the website. I'll update to `develop-2024-06-16` then, although `develop-2024-06-23` is still missing in action.

Reply:
`develop-2024-06-16` and `develop-2024-06-09` trigger the following problem:
```
==> Error: Failed to install gcc-runtime due to NoVerifyException: Spack found new style signed binary packages, but was unable to verify any of them.  Please obtain and trust the correct public key.  If these are public spack binaries, please see the spack docs for locations where keys can be found.
```

Reverting to `develop-2024-06-02` works. I'm not yet sure if this is related to the missing tags from the web UI.

Reply:
I'm going to merge this as-is, and re-visit the issue with Spack's build cache later.